src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachExceptionRecordToExistingCaseTest.java

```diff
@@ -580,7 +584,99 @@ public void should_fail_when_classification_is_supplementary_evidence_with_ocr_d
                     RESPONSE_FIELD_ERRORS,
                     hasItem("The 'attach to case' event is not supported for supplementary evidence with OCR "
                         + "but not containing OCR data")
-        );
+                );
+    }
+
+    @Test
+    public void should_fail_when_classification_is_exception_and_exception_record_contains_pending_payments() {
+        CallbackRequest callbackRequest =
+            callbackRequestWith(
+                EVENT_ID_ATTACH_TO_CASE,
+                CLASSIFICATION_EXCEPTION,
+                "Yes", // awaiting payments DCN processing
+                false
+            );
+
+        given()
+            .body(callbackRequest)
+            .headers(userHeaders())
+            .post(CALLBACK_ATTACH_CASE_PATH)
+            .then()
+            .statusCode(200)
+            .body(
+                RESPONSE_FIELD_ERRORS,
+                hasItem("Cannot attach this exception record to a case because it has pending payments")
+            );
+    }
+
+    @Test
+    public void should_not_return_payments_error_when_supplementary_evidence_with_ocr_have_pending_payments() {
+        CallbackRequest callbackRequest =
+            callbackRequestWith(
+                EVENT_ID_ATTACH_TO_CASE,
+                SUPPLEMENTARY_EVIDENCE_WITH_OCR.name(),
+                "Yes", // awaiting payments DCN processing
+                true
+            );
+
+        given()
+            .body(callbackRequest)
+            .headers(userHeaders())
+            .post(CALLBACK_ATTACH_CASE_PATH)
+            .then()
+            .statusCode(200)
+            .body(
+                RESPONSE_FIELD_ERRORS,
+                not(hasItem("Cannot attach this exception record to a case because it has pending payments"))
+            );
+    }
+
+    @Test
+    public void should_fail_when_classification_is_supplementary_evidence_with_pending_payments() {
+        CallbackRequest callbackRequest =
+            callbackRequestWith(
+                EVENT_ID_ATTACH_TO_CASE,
+                SUPPLEMENTARY_EVIDENCE.name(),
+                "Yes", // awaiting payments DCN processing
+                false
+            );
+
+        given()
+            .body(callbackRequest)
+            .headers(userHeaders())
+            .post(CALLBACK_ATTACH_CASE_PATH)
+            .then()
+            .statusCode(200)
+            .body(
+                RESPONSE_FIELD_ERRORS,
+                hasItem("Cannot attach this exception record to a case because it has pending payments")
+            );
+    }
+
+    @Test
+    public void should_succeed_when_classification_is_supplementary_evidence_with_processed_payments() {
```

Comment:
this test class now has 1111 lines of code. Maybe it would be worth moving the 'pending payments' tests to a new class?